File: src/bin/0371-sum-of-two-integers.py
# @leetup=info id=371 lang=python3 slug=sum-of-two-integers




# @leetup=code
class Solution:
    def getSum(self, a: int, b: int) -> int:
        mask = (1 << 32) - 1
        while b & mask:
            (a, b) = (a ^ b, (a & b) << 1)
        return (a & mask) if b else a


# @leetup=code

solution = Solution()
print(solution.getSum(1, 2))
print(solution.getSum(-1, 1))

File: src/bin/0371-sum-of-two-integers.rs
// @leetup=info id=371 lang=rust slug=sum-of-two-integers

// @leetup=code

impl Solution {
    pub fn get_sum(mut a: i32, mut b: i32) -> i32 {
        while b != 0 {
            // (a, b) = (a ^ b, (a & b) << 1)
            let (a1, b1) = (a ^ b, (a & b) << 1);
            a = a1;
            b = b1;
        }
        a
    }
}
// @leetup=code

struct Solution;

fn main() {
    let _ = Solution::get_sum;
}
